airflow-core/src/airflow/jobs/job.py

```diff
@@ -247,7 +247,9 @@ def heartbeat(
                     session.merge(self)
                     self.latest_heartbeat = timezone.utcnow()
                     session.commit()
-                    time_since_last_heartbeat = (timezone.utcnow() - previous_heartbeat).total_seconds()
+                    time_since_last_heartbeat: float = 0
+                    if previous_heartbeat is not None:
+                        time_since_last_heartbeat = (timezone.utcnow() - previous_heartbeat).total_seconds()
```

Comment:
nit: this can be a ternary operation to avoid assigning to `time_since_last_heartbeat` twice

```suggestion
                    time_since_last_heartbeat: float = 0 if previous_heartbeat is None else (timezone.utcnow() - previous_heartbeat).total_seconds()
```

Reply:
Done. Thanks for approving!